src/main/scala/ru/org/codingteam/keter/scenes/game/GameScene.scala

```diff
@@ -27,48 +28,74 @@ class GameScene(display: Display, engine: Engine) extends Scene(display) with Lo
     render()
   }
 
+  abstract class PlayerCommand() {
+    def execute(snapshot: UniverseSnapshot, player: Person): Unit
+  }
+
+  case class WaitCommand() extends PlayerCommand() {
+    override def execute(snapshot: UniverseSnapshot, player: Person): Unit = {
+      processAction(WaitAction())
+    }
+  }
+
+  case class MoveCommand(dx: Int, dy: Int) extends PlayerCommand() {
+    override def execute(snapshot: UniverseSnapshot, player: Person): Unit = {
+      val move = Move(dx, dy)
+      val target = player.position.moveWithJumps(move).objectPosition
+      snapshot.findActors(target).headOption match {
+        case None => processAction(WalkAction(move))
+        case Some(actor) =>
+          import ru.org.codingteam.keter.util.castToOption
```

Comment:
I am not the author of this concept BTW, I've just moved this code from some another place.